plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineStatusHandler.java

```diff
@@ -119,6 +120,12 @@ private void handleMachineRunning(final String machineId, final WorkspaceRuntime
         if (machine == null) {
             return;
         }
-        eventBus.fireEvent(new MachineStateEvent(machine, RUNNING));
+
+        try {
+            eventBus.fireEvent(new MachineStateEvent(machine, RUNNING));
```

Comment:
Yeah. You're right. Of course you don't know how much handlers are listening for the event.
Because when you're firing an event you shouldn't worry about amount of handlers. This is responsibility of EventBus